components/compass-runtime-agent/internal/graphql/gql_client_testkit.go

```diff
@@ -9,38 +9,43 @@ import (
 )
 
 type QueryAssertClient struct {
-	t                  *testing.T
-	expectedRequests   []*graphql.Request
-	shouldFail         bool
-	modifyResponseFunc ModifyResponseFunc
+	t             *testing.T
+	responseMocks []ResponseMock
+	shouldFail    bool
 }
 
 type ModifyResponseFunc func(t *testing.T, r interface{})
 
+type ResponseMock struct {
```

Comment:
I scratch my head while trying to figure out why we had that funny counter variable (responseIndex). Thank you for clearing things out